### PR TITLE
Updated test workflow to verify certificate update process

### PR DESCRIPTION
**Pull request by agazeley**
The change enables the testing of both certificate updates

Changes to test procedure:
   - Bump action versions to 'v0.2.0'
  -  Add update of nginx certificate where we modify the virtual paths with the GITHUB_RUN_ID to ensure each run is uniquely verifiable. This is a long lived certificate in our AKV instance 
   - Add additional sed command to update multi file config w/ GITHUB_RUN_ID in path of certificate
   - Add steps for creating cert file from AKV cert and verifying NGINX endpoint can use cert and returns correct header for GITHUB_RUN_ID

Potential Enhancements:
- Use config update and cert update in tandem
- Perform updates to multiple certificates

### CODE REVIEW

File: .github/workflows/testNginxForAzureDeploy.yml [hunk 1]
```diff
@@ -3,63 +3,87 @@
...
         uses: azure/login@v1
         with:
-            client-id: ${{ secrets.AZURE_CLIENT_ID }}
-            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
-            subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-      - name: 'Update config - single file'
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      - name: "Sync NGINX Certificate to NGINX on Azure instance - single cert"
```
Comment (ornj) [inline .github/workflows/testNginxForAzureDeploy.yml:32]: Nit,

```suggestion
      - name: "Sync NGINX Certificate to NGINX for Azure - single cert"
```

Updated product name to match marketplace and removed the word "instance" as it's meaning may be ambiguous.

File: .github/workflows/testNginxForAzureDeploy.yml [hunk 2]
```diff
@@ -3,63 +3,87 @@
...
+          nginx-certificate-details: '[{"certificateName": "$NGINX_CERT_NAME", "keyvaultSecret": "https://$NGINX_VAULT_NAME.vault.azure.net/secrets/$NGINX_CERT_NAME", "certificateVirtualPath": "/etc/nginx/ssl/$GITHUB_RUN_NUMBER/my-cert.crt", "keyVirtualPath": "/etc/nginx/ssl/$GITHUB_RUN_NUMBER/my-cert.key"  } ]'
+
+      - name: "Update config - single file"
         shell: bash
         run: |
           sed -i 's/000000/'"$GITHUB_RUN_NUMBER"'/g' test/configs/single/nginx.conf
           cat test/configs/single/nginx.conf
-      - name: 'Sync NGINX configuration to NGINX on Azure instance - single file'
-        uses: nginxinc/nginx-for-azure-deploy-action@v0
+      - name: "Sync NGINX configuration to NGINX on Azure instance - single file"
```
Comment (ornj) [inline .github/workflows/testNginxForAzureDeploy.yml:45]: Same nit,

```suggestion
      - name: "Sync NGINX configuration to NGINX for Azure - single file"
```

File: .github/workflows/testNginxForAzureDeploy.yml [hunk 3]
```diff
@@ -3,63 +3,87 @@
...
+      - name: "Update config - multi file"
         shell: bash
         run: |
+          sed -i 's/000000/'"$GITHUB_RUN_ID"'/g' test/configs/multi/nginx.conf
+          cat test/configs/single/nginx.conf
           sed -i 's/000000/'"$GITHUB_RUN_ID"'/g' test/configs/multi/conf.d/proxy.conf
           cat test/configs/multi/conf.d/proxy.conf
-      - name: 'Sync NGINX configuration to NGINX on Azure instance - multi file'
-        uses: nginxinc/nginx-for-azure-deploy-action@v0
+      - name: "Sync NGINX configuration to NGINX on Azure instance - multi file"
```
Comment (ornj) [inline .github/workflows/testNginxForAzureDeploy.yml:65]: I know you just fixed the quotes but...

```suggestion
      - name: "Sync NGINX configuration to NGINX for Azure - multi file"
```

File: .github/workflows/testNginxForAzureDeploy.yml [hunk 4]
```diff
@@ -3,63 +3,87 @@
...
           nginx-root-config-file: $NGINX_ROOT_CONFIG_FILE
           transformed-nginx-config-directory-path: $NGINX_TRANSFORMED_CONFIG_DIR_PATH
-      - name: 'Validate config update'
+
+      - name: "Validate config update"
         shell: bash
         run: |
           wget -O - -o /dev/null http://${{ secrets.NGINX_DEPLOYMENT_IP }} | jq '.request.headers."Github-Run-Id"  | test( "'"$GITHUB_RUN_ID"'")'
+      - name: "Create cert file"
+        uses: azure/CLI@v1
```
Comment (ornj) [inline .github/workflows/testNginxForAzureDeploy.yml:80]: Is it any faster to use curl or something to make an authorized request to AKV vs the CLI or hardly worth it?

**Review by ornj — APPROVED** [review]

File: .github/workflows/testNginxForAzureDeploy.yml (hunk 4 above)
Comment (agazeley) [reply .github/workflows/testNginxForAzureDeploy.yml:80]: Not significantly faster no. I found this to be easier and more readable without sacrificing a ton of time waiting to pull images or clone the repository.

**Pull request merged**